[new-product] Add Argo CD

adriens opened the pull request:
# :grey_question: About

[`Argo CD`](https://argo-cd.readthedocs.io/) is a very widely used solution around K8S, GitOPS : it makes it possible to deploy in a very comfortable way.

> Declarative continuous deployment for Kubernetes. 

It gains everyday in traction, so it deserves its page on [`endoflife.date`](https://endoflife.date/) :squid: 

![argocd-ui](https://github.com/endoflife-date/endoflife.date/assets/5235127/a716802a-6e90-497d-b89e-a0f88c5e01d1)

adriens committed:
[new-product] Add "Argo CD"

:squid:  Argo CD is a very widely used solution for CD within K8S GitOPS.

captn3m0 commented on products/argocd.md:
```diff
@@ -0,0 +1,234 @@
+
+auto:
+-   git: https://github.com/jenkinsci/jenkins.git
+    regex: '^v-(?<major>[0-9]+)\.(?<minor>[0-9]+)(\.(?<patch>[0-9]+))?$'
```
default regex should work fine:

```suggestion
-   git: https://github.com/argoproj/argo-cd.git
```

captn3m0 commented on products/argocd.md:
```diff
@@ -0,0 +1,234 @@
+releases:
+
+-   releaseCycle: "2.7"
+    lts: false
```
lts default is false, so we can drop the `lts` key from all releases, as there are no LTS releases.

adriens replied:
> lts default is false, 

Good to know, thanks for ponting this :bulb:

captn3m0 commented on products/argocd.md:
```diff
@@ -0,0 +1,234 @@
+-   releaseCycle: "2.7"
+    lts: false
+    releaseDate: 2023-05-02
+    eol: 
```
```suggestion
    eol: false
```

captn3m0 commented on products/argocd.md:
```diff
@@ -0,0 +1,234 @@
+-   releaseCycle: "2.6"
+    lts: false
+    releaseDate: 2023-02-07
+    eol: 2023-05-02
```
2.6 is supported till the 2.8 release, which hasn't happened yet.
```suggestion
    eol: false
```

captn3m0 commented on products/argocd.md:
```diff
@@ -0,0 +1,234 @@
+    regex: '^v-(?<major>[0-9]+)\.(?<minor>[0-9]+)(\.(?<patch>[0-9]+))?$'
+
+
+releases:
```
Add a comment to easily help calculate eol dates
```suggestion
# eol(X) = releaseDate(X+2)
releases:
```

captn3m0 commented on products/argocd.md:
```diff
@@ -0,0 +1,234 @@
+-   releaseCycle: "2.5"
+    lts: false
+    releaseDate: 2022-10-26
+    eol: 2023-02-07
```
Shouldn't this be 2.7 releaseDate? Or perhaps 2.5.21 releaseDate, since that comes later?

adriens replied:
:bulb: as per `x+2` rule

captn3m0 commented on products/argocd.md:
```diff
@@ -0,0 +1,234 @@
+-   /argocd
+-   /argo
+versionCommand: argocd version
+releasePolicyLink: https://github.com/argoproj/argo-cd/blob/master/SECURITY.md#supported-versions
```
More readable:

```suggestion
releasePolicyLink: https://github.com/argoproj/argo-cd/security/policy#supported-versions
```

captn3m0 commented on products/argocd.md:
```diff
@@ -0,0 +1,234 @@
+    releaseDate: 2018-03-13
+    eol: 2018-04-04
+    latest: "0.1.0"
+    latestReleaseDate: 2018-03-13
```
Lets not include releases prior to 1.0. 

```suggestion
```

adriens replied:
Removed :+1:

captn3m0 commented on products/argocd.md:
```diff
@@ -0,0 +1,234 @@
+
+- [🗓️ Release Process And Cadence](https://argo-cd.readthedocs.io/en/stable/developer-guide/release-process-and-cadence/)
+- [🎫 Argo CD roadmap](https://github.com/orgs/argoproj/projects/25/views/14)
+- [🐦 Argo CD on Twitter](https://twitter.com/argoproj)
```
removed the twitter/roadmap links - they weren't adding much. Instead focus on the policy and cadence.

```suggestion
Only the three most recent minor versions are eligible for patch releases. Versions older than the three most recent minor versions are considered EOL and will not receive bug fixes or security updates.

## [Release Cadence](https://argo-cd.readthedocs.io/en/stable/developer-guide/release-process-and-cadence/)

A minor release occurs four times a year, once every three months. Patch releases occur on an as-needed basis.

captn3m0 reviewed (changes requested):
Not sure about the EOL dates. The policy is also differing in both the pages, so its unclear if `eol(x) = releaseDate(x+2)` or `eol(x) = releaseDate(x+3)`.

I've [raised it in the argo-cd chat](https://cloud-native.slack.com/archives/C01TSERG0KZ/p1690794220925009), lets wait to get a reply before finalizing.

adriens committed:
Update products/argocd.md

Co-authored-by: Nemo <commits@captnemo.in>

adriens committed:
Update argocd.md

as per https://github.com/endoflife-date/endoflife.date/pull/3337#discussion_r1278974679

adriens committed:
Update products/argocd.md

Co-authored-by: Nemo <commits@captnemo.in>

adriens committed:
Update products/argocd.md

Co-authored-by: Nemo <commits@captnemo.in>

adriens committed:
Update products/argocd.md

Co-authored-by: Nemo <commits@captnemo.in>

adriens committed:
Update products/argocd.md

Co-authored-by: Nemo <commits@captnemo.in>

adriens committed:
Update argocd.md

update eol considering eol strategy cf https://github.com/endoflife-date/endoflife.date/pull/3337#discussion_r1279879444

adriens committed:
Update argocd.md

Lets not include releases prior to 1.0. as per of : 
cf https://github.com/endoflife-date/endoflife.date/pull/3337#discussion_r1278983022

adriens committed:
Update products/argocd.md

Co-authored-by: Nemo <commits@captnemo.in>

adriens committed:
Update argocd.md

Applying eol policy (+3)

captn3m0 committed:
[argo] Add identifiers

captn3m0 reviewed (approved):
EOLs look good, I added some identifiers as well. Merging.
